scripts/firebase-deploy

```diff
@@ -77,29 +78,145 @@ get_pr_num() {
 	echo "${pr_num}"
 }
 
-update_links() {
-	local environ="$1" domain="evystage.dev"
+hash_css_js() {
+	local dir="$1" files file hash hashed_file
+	if [[ -n "$(git status --porcelain frontend/hashes.txt)" ]]; then
+		exit_error "hashes.txt is dirty"
+	fi
+	mapfile -t files < <(find "${dir}" -name '*.css' -o -name '*.js')
+	for file in "${files[@]}"; do
+		hash=$(get_hash "${file}")
+		hashed_file="${file%.*}.${hash}.${file##*.}"
+		mv "${file}" "${hashed_file}"
+		ln -s "${hashed_file##*/}" "${file}"
+	done
+}
+
+get_hash() {
+	local file="$1"
+	#  Hashes with 32 bits, or 8 chars[0-9a-f] and 100 file changes in a year
+	#  (cache expiry is one year) has a collision probability of less than
+	#  0.0000000005%.
+	shasum -a 256 "${file}" | cut -c -8
+}
+
+# udpate_refs updates references in hrefs and src attributes and also
+# generates an importmap in all **.html files in the frontend directory.
+update_refs() {
+	local environ="$1" domain="evystage.dev" file htmlfiles
 	if [[ "${environ}" == "prod" ]]; then
 		domain="evy.dev"
 	fi
-
-	echo "update links $domain"
-	find frontend -name "*.html" -exec bash -c 'update_links_in_file "$1" "$2"' _ ${domain} {} \;
+	echo "update hrefs for subdomains on $domain"
+	mapfile -t htmlfiles < <(find frontend -name '*.html')
+	for file in "${htmlfiles[@]}"; do
+		update_refs_in_file "${domain}" "${file}"
+	done
 }
 
-# update_links_in_file updates links in the given HTML file to subdomain,
-# e.g.: href="/docs" → href="https://docs.evy.dev" relevant subdomains are:
-# discord, docs, learn and play.
-update_links_in_file() {
-	local domain="$1" file="$2" target
+# update_refs_in_file updates references in hrefs and src attributes and also
+# generates an importmap in given HTML file.
+update_refs_in_file() {
+	local domain="$1" file="$2" target cssfiles jsfiles modulefiles
 	target="out/firebase/public/${file#frontend/}"
-	sed -E \
-		-e 's (href|value)="/(discord|docs|learn|play) \1="https://\2.'"${domain} g" \
-		"${file}" >"${target}"
-	diff -u "${file}" "${target}"
+
+	# TODO this needs to be reworked for `../css/index.css` and other relative file paths.
+	mapfile -t cssfiles < <(cd out/firebase/public && find css -name '*.*.css')
+	mapfile -t jsfiles < <(cd out/firebase/public && find . -name '*.*.js' | cut -c3-)
+	mapfile -t modulefiles < <(cd out/firebase/public && find ./module -name "*.*.js")
```

Comment:
nit: the quotes around `*.*.js` is inconsistent with the previous two lines (double vs single quotes)